src/rdkafka_mock_cgrp.c

```diff
@@ -1402,19 +1402,65 @@ rd_kafka_mock_cgrp_consumer_member_find_by_instance_id(
 static rd_bool_t rd_kafka_mock_cgrp_consumer_member_subscribed_topic_names_set(
     rd_kafka_mock_cgrp_consumer_member_t *member,
     rd_kafkap_str_t *SubscribedTopicNames,
-    int32_t SubscribedTopicNamesCnt) {
-        rd_bool_t changed = rd_false;
-        if (!SubscribedTopicNames)
+    int32_t SubscribedTopicNamesCnt,
+    const rd_kafkap_str_t *SubscribedTopicRegex) {
+        rd_bool_t changed           = rd_false;
+        rd_list_t *new_subscription = NULL;
+        int32_t i;
+
+        if (!SubscribedTopicNames &&
+            RD_KAFKAP_STR_IS_NULL(SubscribedTopicRegex) &&
+            !member->subscribed_topic_regex)
                 /* When client is sending NULL,
                  * its subscription didn't change */
                 return changed;
 
-        int32_t i;
-        rd_list_t *new_subscription = rd_list_new(
-            SubscribedTopicNamesCnt > 0 ? SubscribedTopicNamesCnt : 1, rd_free);
-        for (i = 0; i < SubscribedTopicNamesCnt; i++) {
-                rd_list_add(new_subscription,
+        if (!RD_KAFKAP_STR_IS_NULL(SubscribedTopicRegex)) {
```

Comment:
This `if` can be changed because of previous assert.
We can check `subscribed_topic_regex` here, if it's different we replace it with a copy of `SubscribedTopicRegex`. Even if `SubscribedTopicRegex` is an empty string it must be replaced.